test/framework/vsphere.go

```diff
@@ -128,230 +128,95 @@ func NewVSphere(t *testing.T, opts ...VSphereOpt) *VSphere {
 	return v
 }
 
-// WithRedHat123VSphere vsphere test with redhat 1.23.
-func WithRedHat123VSphere() VSphereOpt {
+// withVSphereKubeVersionAndOS returns a VSphereOpt that adds API fillers to use a vSphere template for
+// the specified OS family and version (default if not provided), corresponding to a particular
+// Kubernetes version, in addition to configuring all machine configs to use this OS family.
+func withVSphereKubeVersionAndOS(osFamily anywherev1.OSFamily, kubeVersion anywherev1.KubernetesVersion, osVersion ...string) VSphereOpt {
```

Comment:
IMO we should move away from these. It has proved to be difficult to work with
Can't we use `WithKubeVersionAndOS` instead?

Reply:
The provider.KubeVersionAndOS returns a `ClusterConfigFiller` object whereas this field returns a `<provider>Opt` object. Also the methods used within are specific to each provider. I do agree that in the future, it'd definitely be better to abstract it to a common method and just inject the provider for this method to do the right thing. But I think in the interest of time, we should tackle this as a follow-up. I did refactor each provider framework by adding common methods to reduce duplication, but I can take this as a follow-up.

Reply:
maybe I missed something
Is this method covering existing code or did you add it new?
if I'm not making sense, let's move this offline

Reply:
It's existing code, just changed the function signature and unexported it.